### PR TITLE
Tool shortcuts

### PR DESCRIPTION
**Pull request by iffn**
Implemented tool specific shortcuts:
- The different shapes now have the ability to define their own shortcuts
- The `EditShape` class has an initial shortcut combination, where holding S and scrolling changes the size.
- New derived shapes can add expand the shortcuts redefine all.
- The shortcut functions have been implemented as classes which can be reused.
- Each class also has a string description that is automatically displayed.

Additional additions:
- Press escape to cancel editing.

To note about the implementation:
- This currently focuses on the `SimpleClickToModifyTool` implementation.
- `[ExecuteInEditMode]` had to be added to `EditShape` in order to make it work.
- Each `ShortcutHandler` class has the ability to take the scene focus. This disables the ability to use the rest of the Unity UI as long as the tool is active. Otherwise, the scene would first have to be activated (For example by right clicking) in order to make the shortcut work.

### CODE REVIEW

File: Source/Edit tool scripts/EditShape.cs [hunk 1]
```diff
@@ -46,6 +48,42 @@ public float OptimizedDistance(Vector3 worldPoint)
...
+        readonly List<ShortcutHandler> shortcutHandlers = new List<ShortcutHandler>();
+
+        void OnEnable()
+        {
+            // ToDo: Find better way to set it up once
+            shortcutHandlers.Clear();
+            SetupShortcutHanlders();
+        }
+
+        protected virtual void SetupShortcutHanlders()
```
Comment (danielfvm) [inline Source/Edit tool scripts/EditShape.cs]: Typo: *SetupShortcutHanlders*

File: Source/Edit tool scripts/ShortcutHandler.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+    public abstract string ShortcutText { get; }
+    public abstract void HandleShortcut(Event e);
+
+    protected void ForceSceneFocus()
+    {
+        if (SceneView.lastActiveSceneView != null)
+        {
+            SceneView.lastActiveSceneView.Focus(); // Make sure that the scene is the focus
+        }
+    }
```
Comment (iffn) [inline Source/Edit tool scripts/ShortcutHandler.cs]: This seems to cause more problems that it solves.

File: Source/Edit tool scripts/EditShape.cs (hunk 1 above)
Comment (iffn) [reply Source/Edit tool scripts/EditShape.cs]: Fixed

File: Source/Edit tool scripts/ShortcutHandler.cs (hunk 2 above)
Comment (iffn) [reply Source/Edit tool scripts/ShortcutHandler.cs]: Removed and replaced with hint.

File: Source/Edit tool scripts/EditShape.cs [hunk 3]
```diff
@@ -46,6 +48,43 @@ public float OptimizedDistance(Vector3 worldPoint)
...
         [SerializeField] private Material linkedMaterial;
 
+        readonly List<ShortcutHandler> shortcutHandlers = new List<ShortcutHandler>();
+
+        void OnEnable()
+        {
+            // ToDo: Find better way to set it up once
+            shortcutHandlers.Clear();
+            SetupShortcutHandlers();
+        }
```
Comment (iffn) [inline Source/Edit tool scripts/EditShape.cs]: Replace with setup function, since this only works on scene objects.

Comment (iffn) [reply Source/Edit tool scripts/EditShape.cs]: Addressed, but not ideally.

**Comment by iffn** [comment]
This has been further improved with the heightmap shapes.
https://github.com/iffn/MarchingCubeEditorForUnity/pull/9

**Pull request merged**